docs/storage.md

```diff
@@ -69,6 +74,11 @@ There's 2 types of storage:
 Cost of thick provisioned is relatively high when you do snapshots (used for backup). If you can use ext or NFS instead, you'll save a LOT of space.
 :::
 
+:::tip
+ZFS provides thick provisioning using the quota properties and thin provisioning using reserved space:
+https://docs.oracle.com/cd/E23823_01/html/819-5461/gazvb.html
```

Comment:
I don't get this. So far, our ZFS implem is very simple: we use the SMAPIv1 file driver that will use VHD files (as for local ext). So we just create the ZFS, then use file on top of the mount point. There's no thick thing anywhere?

Reply:
We have this driver in 8.2: https://github.com/xcp-ng/sm/commit/85121aa11ca943a42aa3d284859c6b383215ccfa
This driver uses a ZFS mountpoint directly created by the user. The driver itself doesn't create the ZFS volume. Of course you can't modify the provisioning using the sm driver but any user can modify the quota properties using the zfs binaries.

Reply:
Okay but it's confusing to me. Thin is the default behavior, right? How it's relevant here? This tip sections opens a lot of questions, far more than answers. If the user is advanced enough, he'll know what to do with its ZFS set. Otherwise, it will be confusing. However, we can add some steps on how to create a ZFS volume manually (something that we'll add via a XAPI plugin later I think)

Reply:
Okay, I agree. If the user knows how to use ZFS, he will know what to do. Let's keep the documentation simple.